src/app/inventory/item-move-service.ts

```diff
@@ -1075,32 +1082,53 @@ export function sortMoveAsideCandidatesForStore(
   fromStore: DimStore,
   targetStore: DimStore,
   getTag: (item: DimItem) => TagValue | undefined,
+  isInInGameLoadoutFor: (item: DimItem, ownerId: string) => boolean,
   /** The item we're trying to make space for. May be missing. */
-  item?: DimItem
+  displacer?: DimItem
 ) {
   // A sort for items to use for ranking *which item to move*
   // aside. The highest ranked items are the most likely to be moved.
   // Note that this is reversed, so higher values (including true over false)
   // come first in the list.
   const itemValueComparator: (a: DimItem, b: DimItem) => number = reverseComparator(
     chainComparator(
+      // MECHANICAL CONSIDERATIONS RELATED TO MOVING ITEMS
+
       // Try our hardest never to unequip something
-      compareBy((i) => !i.equipped),
-      // prefer same bucket over everything
-      compareBy((i) => item && i.bucket.hash === item.bucket.hash),
-      // or at least same category (weapons, armor)
-      compareBy((i) => item && i.bucket.sort === item.bucket.sort),
-      // Always prefer keeping something that was manually moved where it is
-      compareBy((i) => -getLastManuallyMoved(i)),
+      compareBy((displaced) => !displaced.equipped),
+      // prefer same bucket over everything, because that makes space in the correct "pocket"
+      compareBy((displaced) => displacer && displaced.bucket.hash === displacer.bucket.hash),
```

Comment:
I suppose this doesn't need to be a constraint if the `fromStore` is the vault.

Reply:
agreed and that would be a big improvement for once this algo also takes into account amount of free space